### ISSUE TITLE
Wrong type definition for Pagination

### ISSUE THREAD

**Issue by cyyyu** [eleme/element-react#510]
### Description

Just encountered the issue.

<img width="794" alt="wx20170724-173704 2x" src="https://user-images.githubusercontent.com/15100664/28517506-2cd08a0a-7097-11e7-9c4b-699536b4a893.png">

Using `layout="total, prev, pager, next"` as recomended([docs](https://eleme.github.io/element-react/#/en-US/pagination)) is wrong and will be fail to pass typescript type checking.

I would say this api is designed too concise. Something like 

```
interface paginationLayout: {
  prev?: boolean
  next?: boolean
  total?: boolean
}
```

could be better. Just my opinion.

**Comment by e1emeb0t** [comment]
I agree, but the design and most of the code are borrowed from `element-ui`, this makes the usage keep the same as possible.

**Comment by cyyyu** [comment]
ok then.
to solve this issue for now and keep the original api design as well, my proposal is easily having `layout?: string`.
so that we can pass the ts type checking

**Comment by e1emeb0t** [comment]
I'm not familiar with TypeScript, I'll ask someone to solve this issue, thanks.

**Comment by cyyyu** [comment]
thanks~
